internal/sim/generators.go

```diff
@@ -43,6 +35,109 @@ func Flip(p float64) Generator[bool] {
 	})
 }
 
+// Numerics
+
+// Nat returns a [Generator] that returns non-negative numbers, also known as
+// natural numbers. Note that Nat does not return all natural numbers
+// equiprobably. Instead, it biases towards numbers closer to zero and other
+// pathological numbers that are more likely to induce bugs (e.g.,
+// math.MaxInt).
+func Nat() Generator[int] {
```

Comment:
Can we rename `Nat` -> `UInt`?

NAT is overloaded and when doing the code review above, I really had no idea what Nat was.

Reply:
I find `UInt` a bit confusing since we're generating `int`s and not `uint`s. I could change the type to return `uint`s but I also found that a bit annoying because I feel like people work with `int`s way more than they do `uint`s. 

Do you like any of these names?

- `Natural`
- `NaturalNumber`
- `NonNegative`
- `NonNeg`

Reply:
We chatted offline and decided on `NonNegativeInt`.